### ISSUE TITLE
xcodebuild action wrong order of clean build

### ISSUE THREAD

**Issue by iONsky** [fastlane/fastlane#1096]
When you use the `xcodebuild` action and set both `clean` and `build` to true xcodebuild first builds the project and then cleans it.

I looked at the output and realised that fastlane calls xcodebuild with `build clean` instead of `clean build` and xcodebuild seems to execute the actions based on there order.

Fastfile:
```ruby
xcodebuild(
    scheme: "Test",
    clean: true,
    build: true
)
```
Output:
> [08:57:45]: ------------------------
> [08:57:45]: --- Step: xcodebuild ---
> [08:57:45]: ------------------------
> [08:57:45]: Using workspace "Test.xcworkspace"
> [08:57:45]: For a more detailed xcodebuild log open > /Users/Test/Library/Logs/fastlane/xcbuild/2016-01-14/60585/xcodebuild.log
 [08:57:45]: [SHELL COMMAND]: set -o pipefail && xcodebuild -scheme "Test" -workspace "Test" **build clean** | tee '/Users/Test/Library/Logs/fastlane/xcbuild/2016-01-14/60585/xcodebuild.log' | xcpretty --color --simple

**Comment by mfurtak** [comment]
I have verified this locally. Thanks for the report! :+1:

**Comment by mfurtak** [comment]
I've merged a somewhat hacky fix in #1120. Despite `clean` and `build` being boolean flags in the parameters you provided, the order in which you specify them will now matter for getting the output you desire. Your current example of `clean: true` followed by `build: true` should work.

Folks should also be aware that we recommend people use [gym](https://github.com/fastlane/gym) and [scan](https://github.com/fastlane/scan) instead of using the `xcodebuild` action directly!
